gslib/utils/wrapped_credentials.py

```diff
@@ -0,0 +1,155 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Classes and functions to allow google.auth credentials to be used within oauth2client."""
+
+import copy
+import datetime
+import io
+import json
+
+from google.auth import aws, credentials, identity_pool
+
+import oauth2client
+from google.auth.transport import requests
+from gslib.utils import constants
+
+# Expiry is stored in RFC3339 UTC format
+EXPIRY_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+
+DEFAULT_SCOPES = [
+    constants.Scopes.CLOUD_PLATFORM,
+    constants.Scopes.CLOUD_PLATFORM_READ_ONLY,
+    constants.Scopes.FULL_CONTROL,
+    constants.Scopes.READ_ONLY,
+    constants.Scopes.READ_WRITE,
+]
+
+
+class WrappedCredentials(oauth2client.client.OAuth2Credentials):
+
+  def __init__(self, base):
+    if not isinstance(base, credentials.Credentials):
+      raise TypeError("Invalid Credentials")
+    self._base = base
+    super(WrappedCredentials, self).__init__(access_token=None,
+                                             client_id=None,
+                                             client_secret=None,
+                                             refresh_token=None,
+                                             token_expiry=None,
+                                             token_uri=None,
+                                             user_agent=None)
+
+  def _do_refresh_request(self, http):
+    self._base.refresh(requests.Request())
+    if self.store is not None:
+      self.store.locked_put(self)
+
+  @property
+  def access_token(self):
+    return self._base.token
+
+  @access_token.setter
+  def access_token(self, value):
+    self._base.token = value
+
+  @property
+  def token_expiry(self):
+    return self._base.expiry
+
+  @token_expiry.setter
+  def token_expiry(self, value):
+    self._base.expiry = value
+
+  def _to_json(self, strip, to_serialize=None):
+    """Utility function that creates JSON repr. of a Credentials object.
+
+    Args:
+        strip: array, An array of names of members to exclude from the
+                JSON.
+        to_serialize: dict, (Optional) The properties for this object
+                      that will be serialized. This allows callers to
+                      modify before serializing.
+
+    Returns:
+        string, a JSON representation of this instance, suitable to pass to
+        from_json().
+    """
+
+    serialized_data = super()._to_json(strip=strip, to_serialize=to_serialize)
+    deserialized_data = json.loads(serialized_data)
+    deserialized_data['_base'] = copy.copy(self._base.info)
```

Comment:
Do you need to add access token and expiry information as well that you had in the previous code?

Reply:
The parent function actually takes care of expiry information, but it looks liek access token should be handled manually

Reply:
Done